### PR TITLE
API-8319: Document new event form

### PR DESCRIPTION
**Pull request by Adimus11**
## 🚀 Links

- [Feature branch](https://developers.labs.livechat.com/docs/feature/API-8319-document-new-event-form)
- [Jira](https://livechatinc.atlassian.net/browse/API-8319)

## ✅ Checklist (for API docs)

- Changelog [x]
- API versions [x]
- Web & RTM [x]

For more guidelines, see [Updating API docs](https://livechatinc.atlassian.net/wiki/spaces/PAT/pages/761529400/Updating+API+docs).

### Content

- New content

### CODE REVIEW

File: content/messaging/agent-chat-api/changelog/index.mdx [hunk 1]
```diff
@@ -34,6 +34,7 @@ This document is the record of all the changes in the **Agent Chat API**, Web an
   - has a new property for visits, `ended_at`.
   - has new properties: `followed`, `online`, `group_ids`, `state`.
 - The [**Form field**](/messaging/agent-chat-api/v3.3/#form-fields) event's `type` property has a new possible value, `subject`. It's a new type that can appear in ticket forms.
+- Added new event [**Form**](/messaging/agent-chat-api/v3.3/#form)
```
Comment (olek-p) [inline content/messaging/agent-chat-api/changelog/index.mdx:37]: I'm guessing @ale-kacperczyk will notice you used past tense in here, while the other entries use present tense

File: content/messaging/agent-chat-api/v3.3/index.mdx [hunk 2]
```diff
@@ -118,6 +118,33 @@ The **File** event informs about an uploaded file.
 </Code>
 </Section>
 
+## Form
+
+<Section>
+<Text>
+
+The **Form** event contains data from a form (sent during agent inactivity for example).
```
Comment (olek-p) [inline content/messaging/agent-chat-api/v3.3/index.mdx:126]: I don't like this phrasing - it's almost the same as for `Filled form` while these 2 events are actually opposite (empty form vs. filled form)

File: content/messaging/agent-chat-api/v3.3/index.mdx [hunk 3]
```diff
@@ -118,6 +118,33 @@ The **File** event informs about an uploaded file.
...
+
+### Response
```
Comment (olek-p) [inline content/messaging/agent-chat-api/v3.3/index.mdx:128]: No `Request` documentation?

File: content/messaging/agent-chat-api/v3.3/index.mdx [hunk 4]
```diff
@@ -118,6 +118,33 @@ The **File** event informs about an uploaded file.
...
+
+| Field        | Returned   | Notes                                                               |
+| ------------ | ---------- | ------------------------------------------------------------------- |
+| `id`         | always     |                                                                     |
+| `custom_id`  | optionally |                                                                     |
+| `created_at` | always     | Date & time format with a resolution of microseconds, `UTC string`. |
+| `type`       | always     |                                                                     |
+| `author_id`  | optionally | If its generated by system then, it's omitted.                      |
```
Comment (olek-p) [inline content/messaging/agent-chat-api/v3.3/index.mdx:136]: Missing apostrophe + wrongly placed comma

File: content/messaging/customer-chat-api/index.mdx [hunk 5]
```diff
@@ -101,6 +101,33 @@ File event informs about a file being uploaded.
 
 </Section>
 
+## Form
```
Comment (olek-p) [inline content/messaging/customer-chat-api/index.mdx:104]: Which version does this file refer to? Isn't it 3.2?

File: payloads/messaging/v3.3/agent-chat-api/events/form.json [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+	"created_at": "2017-10-12T15:19:21.010200Z",
+	"type": "form",
+	"author_id": "b7eff798-f8df-4364-8059-649c35c9ed0c",
+	"properties": {
+		"property_namespace": {
+			"property_name": "property_value"
+		}
+	},
+	"form_id": "1473433500211",
+	"options": [
```
Comment (olek-p) [inline payloads/messaging/v3.3/agent-chat-api/events/form.json:13]: This should be `fields`

**Review by olek-p — CHANGES_REQUESTED** [review]

File: content/messaging/agent-chat-api/v3.3/index.mdx [hunk 7]
```diff
@@ -118,6 +118,42 @@ The **File** event informs about an uploaded file.
 </Code>
 </Section>
 
+## Form
```
Comment (olek-p) [inline content/messaging/agent-chat-api/v3.3/index.mdx:121]: Nothing links here - there is a list of event types higher up on this page where the new event should be referenced

File: content/messaging/agent-chat-api/v3.3/index.mdx [hunk 8]
```diff
@@ -118,6 +118,42 @@ The **File** event informs about an uploaded file.
...
+
+<Section>
+<Text>
+
+The **Form** event contains data of form that can be filled by customer (sent during agent inactivity for example).
+
+### Request
+
+| Field        | Required | Data type | Notes                                                                    |
```
Comment (olek-p) [inline content/messaging/agent-chat-api/v3.3/index.mdx:130]: #opt This table is unnecessarily wide

File: content/messaging/agent-chat-api/v3.3/index.mdx [hunk 9]
```diff
@@ -118,6 +118,42 @@ The **File** event informs about an uploaded file.
...
+| Field        | Returned   | Notes                                                               |
+| ------------ | ---------- | ------------------------------------------------------------------- |
+| `id`         | always     |                                                                     |
+| `custom_id`  | optionally |                                                                     |
+| `created_at` | always     | Date & time format with a resolution of microseconds, `UTC string`. |
+| `type`       | always     |                                                                     |
+| `author_id`  | optionally | If form is generated by `system` then this field is omitted.        |
+| `properties` | optionally |                                                                     |
+| `form_id`    | always     |                                                                     |
+| `fields`     | always     | An array of [form fields](#form-fields)                             |
```
Comment (olek-p) [inline content/messaging/agent-chat-api/v3.3/index.mdx:148]: Watch out, these are not the same fields as filled form fields

**Review by olek-p — CHANGES_REQUESTED** [review]

File: content/messaging/agent-chat-api/v3.3/index.mdx (hunk 9 above)
Comment (ale-kacperczyk) [reply content/messaging/agent-chat-api/v3.3/index.mdx:149]: What are those fields, then? 🤔

**Comment by ale-kacperczyk** [comment]
Sorry for jumping in like that, I realized I wasn't marked as a reviewer only after I've added the changes. What do you think about including the sample use case + image?

File: content/messaging/agent-chat-api/v3.3/index.mdx (hunk 9 above)
Comment (ale-kacperczyk) [reply content/messaging/agent-chat-api/v3.3/index.mdx:149]: + I forgot to delete the description of this field.

File: content/messaging/customer-chat-api/changelog/index.mdx [hunk 10]
```diff
@@ -30,6 +30,7 @@ This document is the record of all the changes in the **Customer Chat API**, Web
   - has a new parameter for images, `alternative_text`,
   - has a new property for buttons, `target`.
 - The [**Form field**](/messaging/customer-chat-api/v3.3/#form-fields) event's `type` property has a new possible value, `subject`. It's a new type that can appear in ticket forms.
+- Introduce new event [**Form**](/messaging/customer-chat-api/v3.3/#form)
```
Comment (olek-p) [inline content/messaging/customer-chat-api/changelog/index.mdx:33]: This is phrased better in the agent-api changelog

**Review by olek-p — APPROVED** [review]
LGTM
Please note I mostly checked the correctness of the feature description. I didn't pay much attention to the phrasing and formatting of the docs - I will leave that to @ale-kacperczyk :)

**Comment by ale-kacperczyk** [comment]
@Adimus11, could you check out me newest changes?

**Comment by Adimus11** [comment]
@ale-kacperczyk  Looks nice :D

**Review by ale-kacperczyk — APPROVED** [review]

**Pull request merged**